### ISSUE TITLE
React does not recognize the `isActive` prop on a DOM element

### ISSUE THREAD

**Issue by adriangiulianotucci** [rafgraph/react-router-hash-link#102]
Hey guys I am struggling with the NavHashLink component. Im using something like this:
```
<NavHashLink smooth activeClassName="active" to="/#work">
     Button
</NavHashLink>
```
And I find two warnings in my console:
1. React does not recognize the `isActive` prop on a DOM element.
2. React does not recognize the `activeClassName` prop on a DOM element.

Can someone help me? Thanks!

**Comment by mqklin** [comment]
@adriangiulianotucci try to use [custom link](https://www.npmjs.com/package/react-router-hash-link#user-content-custom-link) (genericHashLink)

**Comment by adriangiulianotucci** [comment]
> @adriangiulianotucci try to use [custom link](https://www.npmjs.com/package/react-router-hash-link#user-content-custom-link) (genericHashLink)

Hmmm nope I did this and the result is the same:

```
import { NavHashLink } from "react-router-hash-link";
import { genericHashLink } from "react-router-hash-link";

const MyHashLink = genericHashLink(NavHashLink);

<MyHashLink smooth activeClassName="active" to="/#work">
     Button
</MyHashLink>
```

**Comment by mqklin** [comment]
@adriangiulianotucci try this
```
import {Link} from 'react-router-dom';

const MyHashLink = genericHashLink(Link);

<MyHashLink smooth activeClassName="active" to="/#work">
     Button
</MyHashLink>
```

**Comment by adriangiulianotucci** [comment]
> @adriangiulianotucci try this
> 
> ```
> import {Link} from 'react-router-dom';
> 
> const MyHashLink = genericHashLink(Link);
> 
> <MyHashLink smooth activeClassName="active" to="/#work">
>      Button
> </MyHashLink>
> ```

Yes, it does work but I lose the NavHashLink functionality such as activeClassName.

**Comment by mqklin** [comment]
@adriangiulianotucci seems like there is no same function for NavHashLink (`genericNavHashLink`), only for HashLink: https://github.com/rafgraph/react-router-hash-link#custom-link

**Comment by adriangiulianotucci** [comment]
> @adriangiulianotucci seems like there is no same function for NavHashLink (`genericNavHashLink`), only for HashLink: https://github.com/rafgraph/react-router-hash-link#custom-link

Yep that's what I thought. I'll keep this open just in case. Thanks for your time

**Comment by sort72** [comment]
Any updates on this? Having the same issue

**Comment by anuragkanwar** [comment]
I made a workaround :
I am using current location's hash to determine the current active `HashNavLink`

```js
import { useLocation } from "react-router-dom";
import { NavHashLink } from "react-router-hash-link";

const Index = () => {
  const {hash} = useLocation();
  return (
    <div>
        //...
        <NavHashLink
            smooth={true}
            to={"#settings/profile"}
            className={`block px-2 py-3 rounded-lg ${hash === "#settings/profile" ? "bg-default" : ""}`}
          >
            Profile
          </NavHashLink>
         //...
         <div id={"settings/profile"}>Profile</div>
    </div>
  );
};
export default Index;
```
But the **problem** with this approach is, we are using active classes based on route which means every time user scrolls we still have active states on the navlink , so solution could be to use Intersection Observer also along with useLocation() to determine whether link should be active or not.

**Comment by krazykarthik2** [comment]
I have the same issue please resolve 
Error:React does not recognize the `isActive` prop on a DOM element.
it's annoying

**Comment by krazykarthik2** [comment]
I have encountered this issue in many other packages with identical code others having identical code have fixed this by changing isActive to $isActive
